weblate/addons/tests.py

```diff
@@ -1577,3 +1585,302 @@ def test_json(self) -> None:
+
+        translation_a1 = component_a1.translation_set.get(language__code="cs")
+        translation_a2 = component_a2.translation_set.get(language__code="cs")
+        translation_b1 = component_b1.translation_set.get(language__code="cs")
+
+        # delete similar units between components to avoir propagation
+        translation_a1.unit_set.filter(source="Thank you for using Weblate.").delete()
+        translation_a2.unit_set.filter(source="Hello, world!\n").delete()
+
+        self.edit_unit("Hello, world!\n", "Nazdar svete!\n", translation=translation_a1)
+        self.edit_unit(
+            "Thank you for using Weblate.",
+            "Díky za používání Weblate.",
+            translation=translation_a2,
+        )
+        self.edit_unit("Hello, world!\n", "Nazdar svete!\n", translation=translation_b1)
+
+        self.assertEqual(len(resp_a.calls), 2)
+        self.assertEqual(len(resp_b.calls), 1)
+
+    @responses.activate
+    def test_site_wide_scope(self) -> None:
+        """Test webhook addon installed site-wide."""
+        project_b = self.create_project(name="Test 2", slug="project2")
+        component_b1 = self.create_po(
+            new_base="po/project.pot", project=project_b, name="Component B1"
+        )
+
+        WebhookAddon.create(configuration=self.addon_configuration)
+        responses.add(responses.POST, "https://example.com/webhooks", status=200)
+
+        translation_a1 = self.get_translation()
+        translation_b1 = component_b1.translation_set.get(language__code="cs")
+        self.edit_unit("Hello, world!\n", "Nazdar svete!\n", translation=translation_a1)
+        self.edit_unit("Hello, world!\n", "Nazdar svete!\n", translation=translation_b1)
+
+        self.assertEqual(len(responses.calls), 2)
+
+    @responses.activate
+    def test_invalid_response(self):
+        """Test invalid response from client."""
+        self.do_translation_added_test(response_code=301)
+
+    @responses.activate
+    def test_connection_error(self):
+        """Test connection error when during message delivery."""
+        self.do_translation_added_test(body=requests.ConnectionError())
+
+    @responses.activate
+    def test_jsonschema_error(self):
+        """Test payload schema validation error."""
+        with patch(
+            "weblate.addons.webhooks.validate_schema",
+            side_effect=jsonschema.exceptions.ValidationError("message"),
+        ):
+            self.do_translation_added_test(expected_calls=0)
+
+    @responses.activate
+    def test_bulk_changes(self):
+        """Test bulk change create via the propagate() method."""
+        # create another component in project with same units as self.component
+        self.create_po(
+            new_base="po/project.pot", project=self.project, name="Component B1"
+        )
+        # listen to propagate change event
+        self.addon_configuration["events"].append(ActionEvents.PROPAGATED_EDIT)
```

Comment:
You modify a class variable here what might have undesired effects on other tests.